File: implementations/python/node_based.py
class Node:
    def __init__(self, name):
        self.name = name
        self.neighbors = []

    def add_neighbor(self, node):
        self.neighbors.append(node)
        return self

    def __str__(self):
        return f"{self.name} -> {[n.name for n in self.neighbors]}"

File: implementations/go/node_based.go
package main

type Node struct {
	Name      string
	Neighbors []*Node
}

func NewNode(name string) *Node {
	return &Node{
		Name:      name,
		Neighbors: []*Node{},
	}
}

func (n *Node) AddNeighbor(node *Node) *Node {
	n.Neighbors = append(n.Neighbors, node)
	return n
}
